lib/arel_extensions/helpers.rb

```diff
@@ -29,20 +29,23 @@ def self.column_of_via_arel_table(table_name, column_name)
   end
 
   def self.column_of(table_name, column_name)
-    use_arel_table = !ActiveRecord::Base.connected? || \
-      (ActiveRecord::Base.connection.pool.respond_to?(:schema_cache) && ActiveRecord::Base.connection.pool.schema_cache.nil?)
+    begin
+      use_arel_table = !ActiveRecord::Base.connected? || \
+        (ActiveRecord::Base.connection.pool.respond_to?(:schema_cache) && ActiveRecord::Base.connection.pool.schema_cache.nil?)
 
-    if use_arel_table
-      column_of_via_arel_table(table_name, column_name)
-    else
-      if ActiveRecord::Base.connection.pool.respond_to?(:pool_config)
-        ActiveRecord::Base.connection.pool.pool_config.schema_cache.columns_hash(table_name)[column_name]
-      elsif ActiveRecord::Base.connection.pool.respond_to?(:schema_cache)
-        ActiveRecord::Base.connection.pool.schema_cache.columns_hash(table_name)[column_name]
-      else
-        puts ">>> We really shouldn't be here #{table_name}.#{column_name}"
+      if use_arel_table
         column_of_via_arel_table(table_name, column_name)
+      else
+        if ActiveRecord::Base.connection.pool.respond_to?(:pool_config)
+          ActiveRecord::Base.connection.pool.pool_config.schema_cache.columns_hash(table_name)[column_name]
+        elsif ActiveRecord::Base.connection.pool.respond_to?(:schema_cache)
+          ActiveRecord::Base.connection.pool.schema_cache.columns_hash(table_name)[column_name]
+        else
+          column_of_via_arel_table(table_name, column_name)
+        end
       end
+    rescue
```

Comment:
You don't need the begin/end in Ruby.
```
def foo
...
rescue
...
end
```
does what you mean.

I don't think we should discard any error.  I would only catch the ones triggered by `('foo', 'bar')`.

And I'm uneasy with completely dropping errors: we never `warn` or `log` them?

Introducing `pool = ActiveRecord::Base.connection.pool` would be welcome.